hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientReliableTopicProxy.java

```diff
@@ -0,0 +1,333 @@
+import com.hazelcast.client.impl.HazelcastClientInstanceImpl;
+import com.hazelcast.client.spi.ClientProxy;
+import com.hazelcast.core.ExecutionCallback;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.core.HazelcastInstanceNotActiveException;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.core.ITopic;
+import com.hazelcast.core.Member;
+import com.hazelcast.core.Message;
+import com.hazelcast.core.MessageListener;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.monitor.LocalTopicStats;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.nio.serialization.SerializationService;
+import com.hazelcast.ringbuffer.OverflowPolicy;
+import com.hazelcast.ringbuffer.ReadResultSet;
+import com.hazelcast.ringbuffer.Ringbuffer;
+import com.hazelcast.ringbuffer.StaleSequenceException;
+import com.hazelcast.spi.exception.DistributedObjectDestroyedException;
+import com.hazelcast.topic.ReliableMessageListener;
+import com.hazelcast.topic.TopicOverloadException;
+import com.hazelcast.topic.TopicOverloadPolicy;
+import com.hazelcast.topic.impl.reliable.ReliableMessageListenerAdapter;
+import com.hazelcast.topic.impl.reliable.ReliableTopicMessage;
+
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+import static com.hazelcast.ringbuffer.impl.RingbufferService.TOPIC_RB_PREFIX;
+import static com.hazelcast.topic.impl.reliable.ReliableTopicService.SERVICE_NAME;
+import static com.hazelcast.util.Preconditions.checkNotNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class ClientReliableTopicProxy<E> extends ClientProxy implements ITopic<E> {
+    public static final int MAX_BACKOFF = 2000;
+    public static final int INITIAL_BACKOFF_MS = 100;
+
+    protected final ILogger logger = Logger.getLogger(getClass());
+
+    private final ConcurrentMap<String, MessageRunner> runnersMap
+            = new ConcurrentHashMap<String, MessageRunner>();
+
+    private final String name;
+    private final Ringbuffer ringbuffer;
+    private final SerializationService serializationService;
+    private final ClientReliableTopicConfig config;
+    private final Executor executor;
+    private final TopicOverloadPolicy overloadPolicy;
+
+    public ClientReliableTopicProxy(String objectId, HazelcastClientInstanceImpl client) {
+        super(SERVICE_NAME, objectId);
+        this.name = objectId;
+        this.ringbuffer = client.getRingbuffer(TOPIC_RB_PREFIX + name);
+        this.serializationService = client.getSerializationService();
+
+        this.config = client.getClientConfig().getReliableTopicConfig(objectId);
+        this.executor = getExecutor(config, client);
+        this.overloadPolicy = config.getTopicOverloadPolicy();
+        config.setExecutor(Executors.newSingleThreadExecutor());
```

Comment:
That is indeed something old. Hmm.. need to see what happened since I changed it to make use of a context provided executor if none was explicitly set on the config.